website/docs/developer/development-guide/code/integration-tests.md

```diff
@@ -0,0 +1,188 @@
+- All resources, including storage mediums, should be deleted after all tests have been run.
+  :::warning
+  When deleted, KMS keys persist for a minimum of 7 days. In order to run tests more often 
+  than every 7-8 days, a randomized prefix could be used for   deployment to avoid collision
+  errors with KMS key names within the same environment.
+  :::
+- Initially, automated validation will not include checking Cloudwatch logs. Logs will be available for 7 days to help manually identify any errors and troubleshoot problems. In the future, automating searches for key phrases in Cloudwatch logs as validation may be used for identifying point-of-failure in processes.
+- Integration tests should be run on a regular cadence. Initial suggestion is once every 1-2 weeks.
+
+Some broad categories of tests are shown below.
+
+### Happy Path
+
+These are basic tests that perform a full run-through of a feature.
+Randomized data is used to create a valid state, and a valid user is used to trigger/retrieve work.
+
+### Security Paths
+
+If feasible, there should be a copy of each [Happy Path](#happy-path) test that checks the same route, but with one property modified so as to fail security checks.
+For lambdas with properly specified permissions, this will involve calling the lambda with a user that does not have permission to invoke the lambda.
+For components such as databases and S3 buckets, this will involve attempting to retrieve/delete/add data with a user that does not have the appropriate permissions.
+
+:::tip
+To improve maintainability, use a shared function for setup of the [Happy Path](#happy-path) and Security Path tests for a given component.
+:::
+
+:::warning
+Due to how NGAP handles security, even resources with public access enabled cannot be accessed publicly.
+Further research should be done to identify how to perform these tests.
+:::
+
+### Error Paths
+
+These tests verify that when an error is expected, a proper error is returned/raised.
+For example, several entries in our [API](../../../developer/api/api-gateway.md) should properly return a 404 error code if the entry does not exist in the database.
+The test should fail if the API returns the error in a dictionary instead of an HTTP status code, or the API returns any other error code.
+
+## What Not to Test
+
+### External Integrations
+
+We do not presently test integrations with Cumulus or other external consumers.
+As we do not want to deepen coupling with Cumulus, it is best to focus on maintaining a consistent API.
+
+Manual tests should still be run with Cumulus to check for changes in Cumulus output/input schemas, Orca input/output schemas, Cumulus Message Adapter, and the Cumulus Dashboard. These manual tests should replicate the ingest/recovery tests.
+
+### Performance
+
+While there is a desire to eventually develop performance tests, these tests should focus on functionality with generous timeouts.
+
+## Needed Tests
```

Comment:
I think this requires more discussion. Also we should create cards specific to the number of scripts needed. 
Could be a good topic for the architecture meeting